balancer-labs/frontend-v2
# Migrate composables to a provider pattern, so watchers and composables register once.

# Description
The current issue with a lot of token list lag was the fact that our composables were called many times. Given the fact that these composable instances are isolated, they would register their own watchers and computed properties multiple times as well. 
Some of of these watchers and composables are quite hefty in the compute they perform, so it would essentially become a very expensive render every tim.
This PR aims to refactor that logic and move the more used composable contents to a provider pattern, so they are provided once (watchers and computables) registered once and a much needed speed improvement is introduced.

Fixes #TokenListsLagging

## Type of change


## How should this be tested?
Entire app, make sure invest works, withdraw works (balances update when so). Trading works (balances update)

## Checklist:

- [ WIP] I have performed a self-review of my own code
- [X ] I have commented my code where relevant, particularly in hard-to-understand areas
- [ X] My changes generate no new console warnings
